src/group/mls_group/test_mls_group.rs

```diff
@@ -37,3 +43,267 @@ fn test_mls_group_persistence() {
+        )
+        .unwrap();
+
+        // === Alice adds Bob ===
+        let bob_add_proposal = group_alice.create_add_proposal(
+            group_aad,
+            &alice_credential_bundle,
+            bob_key_package.clone(),
+        );
+        let epoch_proposals = &[&bob_add_proposal];
+        let (mls_plaintext_commit, welcome_bundle_alice_bob_option, kpb_option) = group_alice
+            .create_commit(
+                group_aad,
+                &alice_credential_bundle,
+                epoch_proposals,
+                &[],
+                false,
+            )
+            .expect("Error creating commit");
+        let commit = match mls_plaintext_commit.content() {
+            MLSPlaintextContentType::Commit((commit, _)) => commit,
+            _ => panic!("Wrong content type"),
+        };
+        assert!(!commit.has_path() && kpb_option.is_none());
+        // Check that the function returned a Welcome message
+        assert!(welcome_bundle_alice_bob_option.is_some());
+
+        group_alice
+            .apply_commit(&mls_plaintext_commit, epoch_proposals, &[])
+            .expect("error applying commit");
+        let ratchet_tree = group_alice.tree().public_key_tree_copy();
+
+        let group_bob = MlsGroup::new_from_welcome(
+            welcome_bundle_alice_bob_option.unwrap(),
+            Some(ratchet_tree),
+            bob_key_package_bundle,
+        )
+        .unwrap();
+
+        // Make sure that both groups have the same public tree
+        if group_alice.tree().public_key_tree() != group_bob.tree().public_key_tree() {
+            _print_tree(&group_alice.tree(), "Alice added Bob");
+            panic!("Different public trees");
+        }
+
+        // === Bob updates and commits ===
+        let bob_update_key_package_bundle = KeyPackageBundle::new(
+            &[ciphersuite.name()],
+            &bob_credential_bundle,
+            mandatory_extensions.clone(),
+        )
+        .unwrap();
+
+        let update_proposal_bob = group_bob.create_update_proposal(
+            &[],
+            &bob_credential_bundle,
+            bob_update_key_package_bundle.key_package().clone(),
+        );
+        let (mls_plaintext_commit, _welcome_option, _kpb_option) = group_bob
+            .create_commit(
+                &[],
+                &bob_credential_bundle,
+                &[&update_proposal_bob],
+                &[],
+                false, /* force self update */
+            )
+            .unwrap();
+
+        // Now we break Alice's HPKE ciphertext in Bob's commit by breaking
+        // apart the commit, manipulating the ciphertexts and the piecing it
+        // back together.
+        let (commit, confirmation_tag) = match &mls_plaintext_commit.content {
+            MLSPlaintextContentType::Commit((commit, confirmation_tag)) => {
+                (commit, confirmation_tag)
+            }
+            _ => panic!("Bob created a commit, which does not contain an actual commit."),
```

Comment:
`expect()`

Reply:
Again, I'm deconstructing. I did go through and used `expect()`, where I thought I could. But I'm not sure it applies here.